src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -28,7 +28,11 @@ internal class ProjectBuildManager
             { PropertyNames.DesignTimeBuild, bool.TrueString },
 
             // this will force CoreCompile task to execute even if all inputs and outputs are up to date
+#if NETCOREAPP
```

Comment:
So what does this mean if you're using .NET core to load a .NET framework app? Shouldn't this be doing some dynamic conditioning on the targets being loaded rather than a compile-time decision? Put another way, does the .NET-Core-loading-.NET-Framework and .NET-Framework-loading-.NET-Core directions work?

Reply:
This code makes the assumption that when MSBuildWorkspace is used in a .NET Core app, the MSBuild instance it uses will be a .NET Core instance of MSBuild.